Userland/Utilities/mount.cpp

```diff
@@ -197,6 +198,21 @@ static ErrorOr<void> print_mounts()
     return {};
 }
 
+static ErrorOr<void> mount_using_loop_device(int inode_fd, StringView mountpoint, StringView fs_type, int flags)
+{
+    int devctl_fd = TRY(Core::System::open("/dev/devctl"sv, O_RDONLY));
+    int value = inode_fd;
+    TRY(Core::System::ioctl(devctl_fd, DEVCTL_CREATE_LOOP_DEVICE, &value));
+    int loop_device_index = value;
+
+    auto loop_device_path = TRY(String::formatted("/dev/loop/{}", loop_device_index));
+    int loop_device_fd = TRY(Core::System::open(loop_device_path.bytes_as_string_view(), O_RDONLY));
+
+    auto result = Core::System::mount(loop_device_fd, mountpoint, fs_type, flags);
+    TRY(Core::System::ioctl(devctl_fd, DEVCTL_DESTROY_LOOP_DEVICE, &loop_device_index));
```

Comment:
why do we destroy the loop device right away, instead of destroying it on `umount` ?

Reply:
Because we don't really have a way of keep track of which loop device is for which mount, so I just decided that we should destroy it right away.
If we keep the loop device, the user could use it again (which might be undesirable since it leaves a loop device to be used?), but the user could also generate a new loop device for the same file (as was suggested by @gmta, to remove the safeguards for now), so it's up to us to decide if we want to keep the loop device in place so the user could use it again or we just don't care about it and the user needs to create a new loop device.